src/coreclr/src/jit/hwintrinsicxarch.cpp

```diff
@@ -528,40 +534,20 @@ GenTree* Compiler::impBaseIntrinsic(NamedIntrinsic        intrinsic,
         return nullptr;
     }
 
-    unsigned  simdSize = 0;
-    var_types baseType = TYP_UNKNOWN;
-    var_types retType  = JITtype2varType(sig->retType);
-
-    assert(!sig->hasThis());
+    unsigned  simdSize   = 0;
+    var_types sigRetType = JITtype2varType(sig->retType);
 
     if (HWIntrinsicInfo::BaseTypeFromFirstArg(intrinsic))
```

Comment:
This logic stayed just to get the `simdSize`, correct? If so, we might be able to simplify it more by just passing in the already computed `simdSize` (maybe in a follow up PR?)

Reply:
I thought about this but for `NI_Vector128_AsVector128`, we want to use the `simdSize` based on 1st argument (which is `0x20`) and not the one that we got from `lookupSimdSize()` (which is `0x10`). I didn't feel to special case it for just 1 scenario, so calculating it inside `impBaseIntrinsic`.

Reply:
Maybe we could get `simdSize`, pass it through, and then just specialize `NI_Vector128_AsVector128` in here (not important to this PR in any case)?

Reply:
Good point. Done.